### ISSUE TITLE
Add custom headers to requests

### ISSUE THREAD

**Issue by gudarzi** [gudarzi/SaveHere#26]
Add an option for users to add custom headers to their download requests

**Comment by artoodeeto** [comment]
Hello It's me again. When you say custom headers are you talking about this part right here:

![Screenshot 2024-10-04 at 5 01 56 PM](https://github.com/user-attachments/assets/f9be718c-8b19-4ded-b75a-25417bcc3d43)


Basically a way to add custom file name?

**Comment by gudarzi** [comment]
@artoodeeto No, it means any custom headers that users might want to add. Check this link: https://blog.postman.com/what-are-http-headers/

**Comment by artoodeeto** [comment]
Ohh im sorry. lols. as soon as i saw the UI for headers i thought thats what you meant, my bad. Oh yea.  I could add this feature. Do you want the custom header per request so like per row we could do something like:

```
file-to-download | paused 0% | 0 B/s | Use Headers For Filename | X-custom: header 
```

or 

a custom header for all?

**Comment by gudarzi** [comment]
@artoodeeto I think it'd be easier to first implement it for all at once, than to do it per item. We need to let user add a list of headers, input a header name and a header value, and then send it to the backend. The backend then sends all of those headers with each request.

**Comment by artoodeeto** [comment]
ok so. 

step 1: let user create a list custom header for all download items.

step 2: save all custom headers on the backend. 

step 3: before user tries to download anything we must include all the custom headers that was saved by the user. (NOT REQUIRED FOR DOWNLOAD)

step 3.1: for all the consecutive downloads, we include the headers.

**Comment by gudarzi** [comment]
@artoodeeto custom headers must be saved in frontend, not backend. with each download request to the backend, the frontend sends all cutom headers again, to the backend!

**Comment by GordonNg1** [comment]
I'm working on this one as well, just need to do more testing.

**Comment by GordonNg1** [comment]
#62
